src/string_bytes.cc

```diff
@@ -634,7 +581,8 @@ MaybeLocal<Value> StringBytes::Encode(Isolate* isolate,
       }
 
     case ASCII:
-      if (contains_non_ascii(buf, buflen)) {
+      if (simdutf::validate_ascii_with_errors(buf, buflen).error) {
```

Comment:
Any reason using the error version? Is the common case to have invalid ASCII?

Reply:
The `with_errors` variant bails out early if it detects invalid ASCII, instead of running the entire string, so my thought was that it would match the performance profile of the previous code here best.

The common case is to not use this branch at all :)